searchcore/src/vespa/searchcore/proton/server/maintenance_jobs_injector.cpp

```diff
@@ -49,15 +50,15 @@ injectLidSpaceCompactionJobs(MaintenanceController &controller,
                     (calc ? calc->nodeRetired() : false),
                     bucketSpace);
         } else {
-            job = std::make_unique<LidSpaceCompactionJob>(
+            job = std::make_shared<LidSpaceCompactionJob>(
                     config.getLidSpaceCompactionConfig(),
                     std::move(lidHandler), opStorer, fbHandler,
                     diskMemUsageNotifier,
                     config.getBlockableJobConfig(),
                     clusterStateChangedNotifier,
                     (calc ? calc->nodeRetired() : false));
         }
-        controller.registerJobInMasterThread(trackJob(tracker, std::move(job)));
+        controller.registerJobInMasterThread(trackJob(std::move(tracker), std::move(job)));
```

Comment:
Consider not using `std::move(tracker)` here, to ensure `tracker` is valid beyond first iteration of loop.

Reply:
Fixed